ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/common/Uid.java

```diff
@@ -292,18 +299,40 @@ public void print (PrintStream strm)
         strm.print("<Uid:" + this.toString() + ">");
     }
 
-    public String stringForm ()
-    {
-        // no need to synchronize since object is immutable
-
-        if (_stringForm == null)
-            _stringForm = Utility.longToHexString(hostAddr[0]) + Uid.breakChar
-                    + Utility.longToHexString(hostAddr[1]) + Uid.breakChar
-                    + Utility.intToHexString(process) + Uid.breakChar
-                    + Utility.intToHexString(sec) + Uid.breakChar
-                    + Utility.intToHexString(other);
-
-        return _stringForm;
+    public String stringForm () {
+       // no need to synchronize since object is immutable
+
+       if (_stringForm == null) {
+          long[] hostAddr = this.hostAddr;
+          long hostAddr0 = hostAddr[0];
+          long hostAddr1 = hostAddr[1];
+          int process = this.process;
+          int sec = this.sec;
+          int other = this.other;
+          int hostAddr0Chars = Utility.hexCharsOf(hostAddr0);
+          int hostAddr1Chars = Utility.hexCharsOf(hostAddr1);
+          int processChars = Utility.hexCharsOf(process);
+          int secChars = Utility.hexCharsOf(sec);
+          int otherChars = Utility.hexCharsOf(other);
+          int totalChars = hostAddr0Chars + hostAddr1Chars + processChars + secChars + otherChars + 4; // 4 break chars
+          byte[] asciiBytes = new byte[totalChars];
+          int pos = 0;
+          Utility.toHexChars(hostAddr0, asciiBytes, pos, hostAddr0Chars);
+          pos += hostAddr0Chars;
+          asciiBytes[pos++] = (byte) Uid.breakChar;
+          Utility.toHexChars(hostAddr1, asciiBytes, pos, hostAddr1Chars);
+          pos += hostAddr1Chars;
+          asciiBytes[pos++] = (byte) Uid.breakChar;
+          Utility.toHexChars(process, asciiBytes, pos, processChars);
+          pos += processChars;
+          asciiBytes[pos++] = (byte) Uid.breakChar;
+          Utility.toHexChars(sec, asciiBytes, pos, secChars);
+          pos += secChars;
+          asciiBytes[pos++] = (byte) Uid.breakChar;
+          Utility.toHexChars(other, asciiBytes, pos, otherChars);
+          _stringForm = new String(asciiBytes, 0, 0, totalChars);
```

Comment:
@franz1981  I can see this constructor is deprecated, could you use another constructor please?